lib/redis_memo/redis.rb

```diff
@@ -33,6 +34,22 @@ def initialize(
     super([], ring: hash_ring)
   end
 
+  @@script_shas = {}
+  def run_script(script_content, *args)
+    begin
+      script_sha = @@script_shas[script_content]
```

Comment:
hmm, I think this hash look-up might be expensive since the key is the content of an entire file.

Perhaps the run_script abstraction should take both the SHA1 and the raw script; that way the normal happy code path would be faster because we can save the SHA1 somewhere.